### DISCUSSION TITLE
hour(s) to compile bevy?

### DISCUSSION THREAD

**Discussion by s-mayrh** [bevyengine/bevy#9146] [Q&A]
Bevy compiling from scratch takes two hours on my machine and the resulting bevy_dylib.dll has over 700 megabytes. Just now I added the `rand` crate to my tutorial exercise and it took another hour to build the changed dependencies. Is it because I'm on the mingw toolchain? (msvc buildtools are way too large for my 64GB system drive) It happens everytime I add a crate.

Is this normal?

`dev [optimized + debuginfo] target(s) in 59m 41s`

from my Cargo.toml:
```
[dependencies]
bevy = { version = "0.10.0", features = ["dynamic_linking"] }
rand = "0.8.5"

# Enable a small amount of optimization in debug mode
[profile.dev]
opt-level = 1

# Enable high optimizations for dependencies (incl. Bevy), but not for our code:
[profile.dev.package."*"]
opt-level = 3
```

Windows 10.0.1 21H2 9044.3086
LENOVO_MT_80SG_BU_idea_FM_MIIX 310-10ICR
Intel(R) Atom(TM) x5-Z8350 CPU @ 1.44GHz, 1441 MHz, 4 cores
Toolchain: stable-x86_64-pc-windows-gnu (default) rustc 1.70.0 (90c541806 2023-05-31)

**Comment by hymm** [comment]
seems very possible it's normal. bevy takes around 5 minutes to cold compile on my 5600x. Looking at benchmarks your cpu around 10x times slower https://www.cpu-monkey.com/en/compare_cpu-amd_ryzen_5_5600x-vs-intel_atom_x5_z8300.

What are your incremental build times like?

**Reply by kirawi** [reply]
I think your math is a little off :P
It's also likely to me that it would be slowed down just by running Windows, and memory constraints.

**Comment by s-mayrh** [comment]
Usually between 20 and 30 seconds. I was surprised that adding `rand` caused a large part of bevy to be recompiled although I had dynamic linking enabled.

I don't know how many crates I'm going to add to my project while learning bevy, but waiting one hour each time seemed to me like there's something odd with my system configuration (bug?) but I might be wrong on this.

**Reply by Elabajaba** [reply]
I had a tablet with one of those chips ~6 years ago.

It would choke running Chrome or Firefox with just 1 tab at the time. It's just way too underpowered for development, especially with a compiled language, (and especially Rust, which is known for its slow compile times).

With a quad core CPU you also probably need at least 8 gigs of RAM to compile stuff without constantly swapping, and afaik that device came with 2GB?

Even something as basic as a raspberry pi 4 would be almost a doubling of CPU power.
